app/models/orchestration.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "velum/salt"
+require "velum/salt_orchestration"
+
+# Orchestration represents a salt orchestration event
+class Orchestration < ApplicationRecord
+  class OrchestrationAlreadyRan < StandardError; end
+
+  enum kind: [:bootstrap, :upgrade]
+  enum status: [:in_progress, :succeeded, :failed]
+
+  after_create :update_minions
+
+  # rubocop:disable Rails/SkipsModelValidations
+  def run
+    raise OrchestrationAlreadyRan unless jid.blank?
+    update_column :status, Orchestration.statuses[:in_progress]
+    _, job = case kind
+             when "bootstrap"
+               Velum::Salt.orchestrate
+             when "upgrade"
+               Velum::Salt.update_orchestration
+    end
+    update_column :jid, job["return"].first["jid"]
+    true
+  end
+  # rubocop:enable Rails/SkipsModelValidations
+
+  def self.run(kind: :bootstrap)
+    Orchestration.create(kind: kind).tap(&:run)
+  end
+
+  def self.retryable?(kind: :bootstrap)
+    case kind
+    when :bootstrap
+      Orchestration.bootstrap.last.try(:status) == "failed"
+    when :upgrade
+      Orchestration.upgrade.last.try(:status) == "failed"
+    end
+  end
+
+  # Returns the proxy for the salt orchestration
+  def salt
+    @salt ||= Velum::SaltOrchestration.new orchestration: self
+  end
+
```

Comment:
could you add another method that will check if an orchestration is currently ongoing? this would be another safety layer and would be helpful for a bug where i mentioned this PR as a potential fix: https://bugzilla.suse.com/show_bug.cgi?id=1067194#c8 however to fix this bug, velum needs some more adaption

but nevertheless a method like `self.orchestrating?` would enhance further this new model's usefulness

Reply:
i just realized this is already in the status of the model! so the information is already there, even if no orchestration has been run yet. so i leave it up to you if you like to put this in another method